### PR TITLE
Jamie/initial config

### LINKED ISSUE #48: Config: Update AWSGuard to use new config model
As part of this, consider consolidating any utility/helper code in `@pulumi/policy` to make it easy to author other Policy Packs in a similar manner.

Tasks:

#### In `pulumi/pulumi` (PR https://github.com/pulumi/pulumi/pull/4233):

- [x] Add `initialConfig` property to `AnalyzerInfo` gRPC message which is returned when retrieving policy pack metadata. This `initialConfig` represents any config that was specified programmatically to the Policy Pack, for Policy Packs that support programmatic configuration like AWSGuard.

- [x] Take `initialConfig` into account when reconciling default values and any externally specified config.

#### In `pulumi/pulumi-policy` (PR https://github.com/pulumi/pulumi-policy/pull/220):

- [x] Add constructor arg to the `PolicyPack` class that accepts an `initialConfig`.

- [x] Pass the `initialConfig` along with the `GetAnalzyerInfo` gRPC.

#### In `pulumi/pulumi-policy-aws`:

- [x] Rewrite policies to use config system.

- [x] Refactor how we register policies internally.

- [x] Translate the config args that are passed to the `AWSGuard` constructor into a form acceptable for `initialConfig`, and pass the `initialConfig` to the base `PolicyPack` class's constructor (`PolicyPack`).

- [x] Update unit tests to be able to specify config values that will be available to the policy handler's `args.getConfig()`.

### PR DESCRIPTION
**Pull request by jkisk**
Integrate the new Policy configuration model into AWSguard and accompanying unit tests.  Resolves #48

### CODE REVIEW

File: src/security.ts [hunk 1]
```diff
@@ -54,81 +48,75 @@ export interface AcmCertificateExpirationArgs extends PolicyArgs {
...
                 if (!instance.id || instance.status !== "Active") {
                     continue;
                 }
-
                 // Use the AWS SDK to list the access keys for the user, which will contain the key's creation date.
                 let paginationToken = undefined;
-
                 let accessKeysResp: AWS.IAM.ListAccessKeysResponse;
                 do {
                     accessKeysResp = await iam.listAccessKeys({ UserName: instance.user, Marker: paginationToken }).promise();
```
Comment (justinvp) [inline src/security.ts:122]: GH won't let me comment on the line directly, but link 27 has `if (daysSinceCreated > maxKeyAge!) {`. I _think_ the `!` exclamation point can now be removed from `maxKeyAge` after using `Required<T>` above.

File: src/tests/database.spec.ts [hunk 2]
```diff
@@ -104,6 +99,10 @@ describe("#redshiftClusterConfiguration", () => {
                     enable: false,
                 },
                 encrypted: false,
+            }, {
+                clusterDbEncrypted: false,
+                loggingEnabled: false,
+                nodeTypes: false,
```
Comment (justinvp) [inline src/tests/database.spec.ts:105]:
```suggestion
                nodeTypes: [],
```

File: src/tests/database.spec.ts [hunk 3]
```diff
@@ -342,12 +341,7 @@ describe("#rdsInstanceBackupEnabled", () => {
     });
 
     describe("do not check read replicas", () => {
-        const policy = database.rdsInstanceBackupEnabled({
-            enforcementLevel: "mandatory",
-            backupRetentionPeriod: 7,
-            preferredBackupWindow: "window",
-            checkReadReplicas: false,
```
Comment (justinvp) [inline src/tests/database.spec.ts:349]: Should these be config values passed as the new `config` arg to `createResourceValidationArgs` below?

File: src/tests/database.spec.ts (hunk 2 above)
Comment (jkisk) [reply src/tests/database.spec.ts:105]: Thanks for your sharp eye!

File: src/tests/database.spec.ts (hunk 3 above)
Comment (jkisk) [reply src/tests/database.spec.ts:349]: I'm not totally sure where you mean?  I will push up latest commits and perhaps you could take another look?  Thank you!

File: src/tests/database.spec.ts [hunk 4]
```diff
@@ -289,20 +288,20 @@ describe("#dynamodbTableEncryptionEnabled", () => {
...
                 backupRetentionPeriod: 7,
                 backupWindow: "window",
                 replicateSourceDb: "some-there-db",
-            });
+            },
+                {
+                    backupRetentionPeriod: 7,
+                    preferredBackupWindow: "window",
+                    checkReadReplicas: true,
+                });
```
Comment (justinvp) [inline src/tests/database.spec.ts:304]: Nit: indentation

```suggestion
            }, {
                backupRetentionPeriod: 7,
                preferredBackupWindow: "window",
                checkReadReplicas: true,
            });
```

File: src/tests/database.spec.ts [hunk 5]
```diff
@@ -342,20 +341,18 @@ describe("#rdsInstanceBackupEnabled", () => {
...
                 instanceClass: "db.m5.large",
-                backupRetentionPeriod: 0,
                 backupWindow: "another-window",
                 replicateSourceDb: "some-there-db",
-            });
+            },
+                {
+                    backupRetentionPeriod: 7,
+                    checkReadReplicas: true,
+                });
```
Comment (justinvp) [inline src/tests/database.spec.ts:355]: Nit: Indentation

```suggestion
            }, {
                backupRetentionPeriod: 7,
                checkReadReplicas: true,
            });
```

File: src/tests/database.spec.ts [hunk 6]
```diff
@@ -478,7 +464,10 @@ describe("#rdsStorageEncrypted", () => {
                 backupWindow: "random-window",
                 storageEncrypted: true,
                 kmsKeyId: "a-kms-key",
-            });
+            },
+                {
+                    kmsKeyId: "a-kms-key",
+                });
```
Comment (justinvp) [inline src/tests/database.spec.ts:470]: Nit: Indentation

```suggestion
            }, {
                kmsKeyId: "a-kms-key",
            });
```

**Pull request merged**